Remove ancient upgrade considerations

tas50 opened the pull request:
This is totally useless at this point. It's mostly about the server upgrade actually and it's 11 -> 12.

Signed-off-by: Tim Smith <tsmith@chef.io>

kagarmoe commented:
I think the broad outline of the information could be useful to new Chefs. Conceptually, the information seems important in regard to managing node data and cookbook size. I don't think we explain it elsewhere.

How could we revise this section to be about "managing node data and cookbook size" and where could we relocate the content?

tas50 committed:
Remove ancient upgrade considerations

This is totally useless at this point. It's mostly about the server upgrade actually and it's 11 -> 12.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 commented:
None of this matters for a client upgrade though. Things like checking for name in metadata.rb were entirely for the Chef 11 -> 12 server upgrade. We have cookstyle rules for that. We also have cookstyle rules for binaries in the cookbook to keep size down. The recommended methods here of cleaning up old cookbooks are also wrong. This is the kind of doc that is worse than nothing. It takes people down a 2013 path of sadness.

kagarmoe reviewed (approved):
![tenor-6178251](https://user-images.githubusercontent.com/4400151/150597601-16158397-1fcc-42e2-b005-10bfffa91d48.gif)

Commit:
Merge branch 'main' into updates

Commit:
Merge branch 'main' into updates
